exchanges/exchanges.go

```diff
@@ -1154,6 +1318,266 @@ type BittrexCandlestickResponse struct {
+		return -1
+	}
+	return bittrex.queue[0].Nonce
+}
+
+func (bittrex *BittrexExchange) processQueue() {
+	queue := bittrex.queue
+	bittrex.queue = make([]*BittrexOrderbookUpdate, 0)
+	for _, update := range queue {
+		bittrex.processOrderbookUpdate(update)
+	}
+}
+
+// Process an update. Queues the order if it is not sequential.
+func (bittrex *BittrexExchange) processOrderbookUpdate(update *BittrexOrderbookUpdate) {
+	if update.Nonce <= bittrex.orderSeq {
+		// Not necessarily an error. Simply discard.
+		return
+	}
+	if update.Nonce != bittrex.orderSeq+1 {
+		nextNonce := bittrex.queueOrderbookUpdate(update)
+		if nextNonce <= bittrex.orderSeq+1 {
+			bittrex.processQueue()
+		}
+		return
+	}
+	bittrex.orderSeq++
+
+	for _, ask := range update.Sells {
+		bittrex.processBittrexOrderbookPoint(ask, bittrex.asks)
+	}
+	for _, buy := range update.Buys {
+		bittrex.processBittrexOrderbookPoint(buy, bittrex.buys)
+	}
+}
+
+// Handle the initial orderbook from the websocket.
+func (bittrex *BittrexExchange) processFullOrderbook(book *BittrexOrderbookUpdate) {
+	bittrex.orderMtx.Lock()
+	defer bittrex.orderMtx.Unlock()
+	bittrex.buys = translateBittrexOrderbook(book.Buys)
+	bittrex.asks = translateBittrexOrderbook(book.Sells)
+	bittrex.orderSeq = book.Nonce
+	bittrex.processQueue()
+	bittrex.queue = make([]*BittrexOrderbookUpdate, 0)
+	state := bittrex.state()
+	if state != nil { // Only send update if price has been fetched
+		depth := bittrex.wsDepthSnapshot()
+		bittrex.Update(&ExchangeState{
+			Price:        state.Price,
+			BaseVolume:   state.BaseVolume,
+			Volume:       state.Volume,
+			Change:       state.Change,
+			Depth:        depth,
+			Candlesticks: state.Candlesticks,
+		})
+	}
+	bittrex.wsUpdated()
+}
+
+// Handles an update to the orderbook.
+func (bittrex *BittrexExchange) processNextUpdate(update *BittrexOrderbookUpdate) {
+	bittrex.orderMtx.Lock()
+	defer bittrex.orderMtx.Unlock()
+	if bittrex.orderSeq == 0 { // initial orderbook has not been received yet.
+		if !bittrex.obRequested {
+			bittrex.requestOrderbook()
+		}
+		bittrex.queueOrderbookUpdate(update)
+		return
+	}
+	bittrex.processOrderbookUpdate(update)
+}
+
+// Handle the SignalR message. The message can be either a full orderbook at
+// msg.R (msg.I == "1"), or a list of updates in msg.M[i].A.
+func (bittrex *BittrexExchange) msgHandler(msg signalr.Message) {
+	if msg.I == "1" {
+		book := new(BittrexOrderbookUpdate)
+		// the signalr client has already decoded the order book into an interface{}.
+		// Rather than try to match implicit types, just re-encode to []bytes and
+		// decode into the known structure.
+		bookBytes, err := msg.R.MarshalJSON()
+		if err != nil {
+			log.Errorf("bittrex orderbook re-encode error: %v", err)
+			return
+		}
+		err = json.Unmarshal(bookBytes, book)
+		if err != nil {
+			bittrex.setWsFail(fmt.Errorf("Bittrex order book Unmarshal error: %v", err))
+			return
+		}
+		if book == nil {
```

Comment:
Travis pointed out that this can never be `nil`.  Perhaps checking the nonce or the order slice lengths would indicate if Bittrex returned crap data?